libherokubuildpack/src/inventory.rs

```diff
@@ -31,15 +110,23 @@ impl<V, D, M> Default for Inventory<V, D, M> {
 }
 
 impl<V, D, M> Inventory<V, D, M> {
+    /// Creates a new empty inventory
     #[must_use]
     pub fn new() -> Self {
         Self::default()
     }
 
+    /// Add a new artifact to the in-memory inventory
+    ///
+    /// For removal, you can use the `Vec::retain` method on the artifacts attribute
+    /// to remove artifacts that match a given predicate.
```

Comment:
I needed to delete/reject artifacts in the ruby binary builder https://github.com/heroku/docker-heroku-ruby-builder/blob/c5f2030754560bc8d830053a2935063ab14b491a/ruby_executable/src/bin/ruby_build.rs#L201-L203.

I'm fine removing this as well. It's a little weird that the field is public which allows you to push to the vec, but there's a separate push associated function.

> Otherwise, artifacts attribute should probably be changed to artifacts field

In Ruby they're called attributes i.e. `attr_accessor` also I keep wanting to call associated functions methods. Anywhoo. Deleting this all is fine.